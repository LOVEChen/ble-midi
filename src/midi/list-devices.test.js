/* eslint-env jest */
jest.mock('../util/logger');
jest.mock('midi', () => {
  const midi = jest.fn();
  midi.getPortCount = () => 2;
  midi.getPortName = portNum => `MOCK_DEVICE_${portNum}`;
  midi.openVirtualPort = () => {};
  midi.closePort = () => {};
  midi.__mockBleno__ = true;
  return {
    input: () => midi,
    output: () => midi
  };
});

let mockLog;
let mockMidi;
let listDevices;

beforeEach(() => {
  jest.resetAllMocks();
  mockLog = require('../util/logger').default;
  mockMidi = require('midi');
  listDevices = require('./list-devices').default;
});

[
  undefined,
  [],
  ['invalid'],
  ['input'],
  ['output'],
  ['output', 'input'],
  ['input', 'output'],
  ['input', 'invalid', 'output']
].forEach(types => {
  test(`listDevices(${JSON.stringify(types)})`, () => {
    listDevices(types);
    expect(mockLog.mock).toMatchSnapshot();
    expect(mockMidi.mock).toMatchSnapshot();
  });
});
