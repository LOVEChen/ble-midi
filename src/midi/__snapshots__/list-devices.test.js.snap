// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listDevices(["input","invalid","output"]) 1`] = `
Array [
  Array [
    "
Available Devices (input):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
  Array [
    "
Available Devices (output):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
]
`;

exports[`listDevices(["input","invalid","output"]) 2`] = `undefined`;

exports[`listDevices(["input","output"]) 1`] = `
Array [
  Array [
    "
Available Devices (input):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
  Array [
    "
Available Devices (output):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
]
`;

exports[`listDevices(["input","output"]) 2`] = `undefined`;

exports[`listDevices(["input"]) 1`] = `
Array [
  Array [
    "
Available Devices (input):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
]
`;

exports[`listDevices(["input"]) 2`] = `undefined`;

exports[`listDevices(["invalid"]) 1`] = `Array []`;

exports[`listDevices(["invalid"]) 2`] = `undefined`;

exports[`listDevices(["output","input"]) 1`] = `
Array [
  Array [
    "
Available Devices (output):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
  Array [
    "
Available Devices (input):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
]
`;

exports[`listDevices(["output","input"]) 2`] = `undefined`;

exports[`listDevices(["output"]) 1`] = `
Array [
  Array [
    "
Available Devices (output):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
]
`;

exports[`listDevices(["output"]) 2`] = `undefined`;

exports[`listDevices([]) 1`] = `Array []`;

exports[`listDevices([]) 2`] = `undefined`;

exports[`listDevices(undefined) 1`] = `
Array [
  Array [
    "
Available Devices (input):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
  Array [
    "
Available Devices (output):",
  ],
  Array [
    "0: MOCK_DEVICE_0",
  ],
  Array [
    "1: MOCK_DEVICE_1",
  ],
]
`;

exports[`listDevices(undefined) 2`] = `undefined`;
